File: src/denilson/fracoes.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# vi:ts=4 sw=4 et

n = int(raw_input())
for teste in xrange(n):
    nums = [int(x) for x in raw_input().split()]
    if len(nums) != 4:
        print "Erro no teste %d: foram lidos %d numeros!" % (teste, len(nums))

    for i in nums:
        if not 0 < i <= 100:
            print "Erro no teste %d: numero %d fora do intervalo!" % (teste, i)

    a,b,c,d = nums
    if a*d == b*c:
        print "SIM"
    else:
        print "NAO"

File: src/cabessa/fracoes.c
/**/

#include <stdio.h>
#include <string.h>
#include <assert.h>


int main()
{
	int n;
	int a, b, c, d;

	scanf("%d", &n);	

	assert(n>0);

	while (n--)
	{
		scanf("%d %d %d %d", &a, &b, &c, &d);

		assert(a >= 1 && a <= 100);
		assert(b >= 1 && b <= 100);
		assert(c >= 1 && c <= 100);
		assert(d >= 1 && d <= 100);

		if (a*d == b*c)
		{
			printf("SIM\n");
		}
		else
		{
			printf("NAO\n");
		}

	}
	return 0;
}
